src/scatter_buildblock/ScatterEstimation.cxx

```diff
@@ -0,0 +1,1486 @@
+                                                                   this->input_projdata_2d_sptr->get_proj_data_info_sptr()->create_shared_clone(),
+                                                                   out_filename,
+                                                                   std::ios::in | std::ios::out | std::ios::trunc));
+            SSRB(*this->back_projdata_2d_sptr,
+                 *this->back_projdata_sptr, false);
+//            {
+//                std::string out_filename = extras_path.get_path() + "tmp_background_data.hs";
+
+//                this->back_projdata_3d_sptr.reset(new ProjDataInterfile(this->input_projdata_sptr->get_exam_info_sptr(),
+//                                                                        this->input_projdata_sptr->get_proj_data_info_sptr()->create_shared_clone(),
+//                                                                        out_filename,
+//                                                                        std::ios::in | std::ios::out | std::ios::trunc));
+
+//                this->back_projdata_3d_sptr->fill(*this->add_projdata_3d_sptr);
+//            }
+//            {
+//                std::string out_filename = extras_path.get_path() + "tmp_background_data" + "_2d.hs";
+
+//                this->back_projdata_2d_sptr.reset(new ProjDataInterfile(this->input_projdata_2d_sptr->get_exam_info_sptr(),
+//                                                                        this->input_projdata_2d_sptr->get_proj_data_info_sptr()->create_shared_clone(),
+//                                                                        out_filename,
+//                                                                        std::ios::in | std::ios::out | std::ios::trunc));
+//                this->back_projdata_2d_sptr->fill(*this->add_projdata_2d_sptr);
+//                this->multiplicative_binnorm_2d_sptr->apply(*this->back_projdata_2d_sptr, start_time, end_time);
+//            }
+
+        }
+        else
+        {
+            this->back_projdata_2d_sptr = back_projdata_sptr;
+        }
+#else
+        {
+            FilePath tmp(this->back_projdata_filename);
+            std::string in_filename = extras_path.get_path() + tmp.get_filename_no_extension() + "_2d.hs";
+            this->back_projdata_2d_sptr = ProjData::read_from_file(in_filename);
+        }
+#endif
+    }
+    else // We will need a background for the scatter, so let's create a simple empty ProjData
+    {
+        if (run_in_2d_projdata)
+        {
+
+
+            this->back_projdata_2d_sptr.reset(new ProjDataInterfile(this->input_projdata_2d_sptr->get_exam_info_sptr(),
+                                                                    this->input_projdata_2d_sptr->get_proj_data_info_sptr()->create_shared_clone(),
+                                                                    out_filename,
```

Comment:
`out_filename` not set, except ages ago. Are you sure it'll be correct?

Reply:
This line has been commented out and probably get deleted soon.